### ISSUE TITLE
The UI5 Language Assistant server crashed 5 times in the last 3 minutes

### ISSUE THREAD

**Issue by mydoghasfleas** [SAP/ui5-language-assistant#732]
Here is the full log: 
[ui5 lang server log.txt](https://github.com/user-attachments/files/17242963/ui5.lang.server.log.txt)

I haven't tried this in other contexts, but I am creating an adaptation project and trying the autocomplete in a fragment that was added with the adaptation editor, so maybe that is related to the issue?

```
TypeError: Cannot read properties of undefined (reading 'type')
    at SSe (c:\Users\user01\.vscode\extensions\saposs.vscode-ui5-language-assistant-4.0.73\node_modules\@ui5-language-assistant\language-server\dist\server.js:456:16868)
    at lKt (c:\Users\user01\.vscode\extensions\saposs.vscode-ui5-language-assistant-4.0.73\node_modules\@ui5-language-assistant\language-server\dist\server.js:813:36419)
    at Kve (c:\Users\user01\.vscode\extensions\saposs.vscode-ui5-language-assistant-4.0.73\node_modules\@ui5-language-assistant\language-server\dist\server.js:1404:6163)
    at c:\Users\user01\.vscode\extensions\saposs.vscode-ui5-language-assistant-4.0.73\node_modules\@ui5-language-assistant\language-server\dist\server.js:1411:4961
```

**Comment by mydoghasfleas** [comment]
I am getting the exact same issue in SAP Business Application studio. I guess I should open an incident with SAP about it but, as this is where it's being developed, it will probably come back here...

```
TypeError: Cannot read properties of undefined (reading 'type')
    at SSe (/extbin/local/openvscode-server/extensions/vscode-ui5-language-assistant-4.0.73/node_modules/@ui5-language-assistant/language-server/dist/server.js:456:16868)
    at lKt (/extbin/local/openvscode-server/extensions/vscode-ui5-language-assistant-4.0.73/node_modules/@ui5-language-assistant/language-server/dist/server.js:813:36419)
    at Kve (/extbin/local/openvscode-server/extensions/vscode-ui5-language-assistant-4.0.73/node_modules/@ui5-language-assistant/language-server/dist/server.js:1404:6163)
    at /extbin/local/openvscode-server/extensions/vscode-ui5-language-assistant-4.0.73/node_modules/@ui5-language-assistant/language-server/dist/server.js:1411:4961

Node.js v20.15.1
[Error - 11:13:20 AM] The UI5 Language Assistant server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
```

**Comment by MN-public** [comment]
@mydoghasfleas 
Thanks for reporting the issue, we are looking into it. Did you experience it with previous versions too? I.e. is it a regression?
BTW: this is the correct place to report issues, as UI5 language assistant is open source software

**Comment by mydoghasfleas** [comment]
@MN-public I have not tried with any previous version. I have tried creating an app before with the Fiori tools, and I seem to remember content assist working there. But this is the first time with an adaptation project.

**Comment by MN-public** [comment]
@mydoghasfleas Thanks for the details. We identified an issue and prepared a fix. It would resolve the regression introduced for adaptation projects recently.

You can test it by downloading the latest [release 4.0.74](https://github.com/SAP/ui5-language-assistant/releases/download/v4.0.74/vscode-ui5-language-assistant-4.0.74.vsix) as .vsix from GitHub and installing it manually in your dev environment or wait until it is available in SAP Business application studio and VSC marketplace.
We will consider better support of adaptation projects in future

**Comment by mydoghasfleas** [comment]
@MN-public Excellent, thanks, I'll give it a try!

**Comment by mydoghasfleas** [comment]
So, trying it now in an adaptation project, after install the .vsix, the language server now shows as ready.

My annotation XML file is identified with filetype 'edmx' in the status bar.

When pressing Ctrl+Space in the empty file, there is some processing, and then nothing:

![image](https://github.com/user-attachments/assets/9fd2c33b-4f00-4a7d-9d18-795aa758d16e)

Should I be expecting something here at this point?

Even when I add some boilerplate edmx content and try Ctrl+Space in the Schema element, nothing happens.

Should there be some boilerplate generated by the plugins for the annotation? Should I start with a brand new project?

**Comment by MN-public** [comment]
@mydoghasfleas UI5 language assistant is supposed to work with the .fragment.xml and view.xml, not the annotation files. LSP for annotation files is provided by the [SAP Fiori tools - XML annotation language server](https://marketplace.visualstudio.com/items?itemName=SAPSE.sap-ux-annotation-modeler-extension) extension. I'believe it never worked for the adaptation projects, but we are looking to enable it in future.
But in any case, to work with the LSP in the annotation file you need to have some initial content with references to the OData vocabularies and metadata.

**Comment by mydoghasfleas** [comment]
@MN-public Thanks for the info. Is there some documentation to do this? I don't know how to get started with the annotation boilerplate.

**Comment by mydoghasfleas** [comment]
I guess what I'll have to do is create a normal app project first, make some annotations (if I can) with the XML annotation language server, try get that working, then copy and paste that to my adaptation project.
That is about as straightforward as SAP software gets, I guess.

**Comment by MN-public** [comment]
@mydoghasfleas 
Unfortunately there is a gap in the tools support for editing annotation files in adaptation projects at the moment, sorry for that. We are aware of it and working on closing this gap. You can watch the [change log](https://marketplace.visualstudio.com/items/SAPSE.sap-ux-annotation-modeler-extension/changelog) of [SAP Fiori Tools - XML Annotation Language Server](https://marketplace.visualstudio.com/items?itemName=SAPSE.sap-ux-annotation-modeler-extension&ssr=false#overview) for updates, once annotation LSP (IntelliSense) features are provided for annotation files within the adaptation projects, change log will be updated accordingly.

At this moment, the initial issue you reported for UI5 Language Assistant is fixed and the version 4.0.74 containing this fix is released to Visual Studio Code and SAP Business Application Studio, so I'm closing this ticket

**Comment by mydoghasfleas** [comment]
@MN-public thanks for the info.
